src/benchmarks/micro/corefx/System.Text.Json/Utf8JsonWriter/ArrayBufferWriter.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Diagnostics;
+
+namespace System.Text.Json
+{
+    internal sealed class ArrayBufferWriter<T> : IBufferWriter<T>
+    {
+        private T[] _buffer;
+        private int _index;
+
+        private const int MinimumBufferSize = 256;
+
+        public ArrayBufferWriter()
+        {
+            _buffer = new T[MinimumBufferSize];
+            _index = 0;
+        }
+
+        public ArrayBufferWriter(int initialCapacity)
+        {
+            if (initialCapacity <= 0)
+                throw new ArgumentException(nameof(initialCapacity));
+
+            _buffer = new T[initialCapacity];
+            _index = 0;
+        }
+
+        public ReadOnlyMemory<T> WrittenMemory
+        {
+            get
+            {
+                return _buffer.AsMemory(0, _index);
+            }
+        }
+
+        public int WrittenCount
+        {
+            get
+            {
+                return _index;
+            }
+        }
+
+        public int Capacity
+        {
+            get
+            {
+                return _buffer.Length;
+            }
+        }
+
+        public int FreeCapacity
+        {
+            get
+            {
+                return _buffer.Length - _index;
+            }
+        }
+
+        public void Clear()
+        {
+            ClearHelper();
+        }
+
+        private void ClearHelper()
+        {
+            Debug.Assert(_buffer != null);
+
+            _buffer.AsSpan(0, _index).Clear();
+            _index = 0;
+        }
+
+        public void Advance(int count)
+        {
+            if (count < 0)
+                throw new ArgumentException(nameof(count));
+
+            if (_index > _buffer.Length - count)
+                ThrowInvalidOperationException(_buffer.Length);
```

Comment:
You are not reusing the function. Could you please move the throw statement hare and remove the function?

Reply:
The intent of this was not about re-use.

Reply:
But I can be more consistent about it's use.